MathlibTest/superscript.lean

```diff
@@ -81,3 +81,48 @@ run_cmd Lean.Elab.Command.liftTermElabM do
   Lean.logInfo <| ← Lean.PrettyPrinter.ppTerm <| ← `(term| ($a)$α:subscript)
 
 end
+
+section delab
+
+open Lean PrettyPrinter.Delaborator SubExpr
+open Mathlib.Tactic.Superscript (subscriptable superscriptable)
+
+private def check_subscript {α : Type} (_ : α) := ()
+private def check_superscript {α : Type} (_ : α) := ()
+
+@[app_delab check_subscript]
+private def delabCheckSub : Delab := withOverApp 2 do
+  let #[_, x] := (← getExpr).getAppArgs | failure
+  let () ← withAppArg <| subscriptable x
+  `("ok")
```

Comment:
Could you replace these with a more realistic case where you actually delaborate into `subscript` notation, and the test show the result is the same as the input?

Reply:
something like
```suggestion
@[app_delab check_subscript]
private def delabCheckSub : Delab := withOverApp 2 do
  let #[_, x] := (← getExpr).getAppArgs | failure
  let x_d ← withAppArg <| delabSubscript
  `(check_expr($x_d:subscript))
```
with a suitable `macro`